File: leetcode/longsubstrnorepeat/longsubstrnorepeat.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        j = 1
        if len(s) == 0:
            return 0
        thislist = list(s[0])
        longest = len(thislist)
        for j in s[1:]:
            longest = max(longest, len(thislist))
            while j in thislist:
                thislist.pop(0)
            thislist.append(j)
        longest = max(longest, len(thislist))
        return longest

File: leetcode/longsubstrnorepeat/longsubstrnorepeat.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int i, j, longest;
        bool match = false;
        
        longest = 0;
        for (i = 0; i < s.length(); i++) {
            match = false;
            for (j = i + 1; j < s.length(); j++) {
                if (s.substr(i,j-i).find_first_of(s[j]) < (j - i)) {
                    longest = max(longest, j - i);
                    match = true;
                    break;
                }
            }
            if (!match) {
                longest = max(longest, j - i);
            }
        }
        return longest;
    }
};
